### PR TITLE
Add new test to validate multi. part pattern match

### PR DESCRIPTION
**Pull request by silverstar194**
Commit Message:
Add new test to validate multi. part pattern match

Additional Description:
Risk Level:
Low. Testing only.

Testing:
Pattern match test.

Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by silverstar194** [comment]
//assign @RyanTheOptimist

File: test/common/router/config_impl_test.cc [hunk 1]
```diff
@@ -9722,6 +9722,41 @@ TEST_F(RouteMatcherTest, PatternMatchInvalidVariableName) {
       EnvoyException, "path_match_policy.path_template /rest/{on==e}/{two} is invalid");
 }
 
+TEST_F(RouteMatcherTest, PatternMatchWildcardMiddleThreePartVariableNamed) {
```
Comment (RyanTheOptimist) [inline test/common/router/config_impl_test.cc:9725]: Great test name!

**Review by RyanTheOptimist — APPROVED** [review]
Nice! Thanks for improving test coverage.

**Pull request merged**